src/mame/capcom/cps2.cpp

```diff
@@ -10821,6 +10915,8 @@ void cps2_state::init_cps2()
 	// Decrypt the game - see machine/cps2crypt.cpp
 	init_cps2crypt();
 	init_cps2nc();
+
+	m_maincpu->space(AS_PROGRAM).install_read_handler(0x8040b0, 0x8040b2, read8sm_delegate(*this, FUNC(cps2_state::cps2_dsw_r)));
```

Comment:
Much better to have that directly in the memory map.  If you have a compelling reason to do that install here anyway, use emu::rw_delegate instead of read8sm_delegate though.  We're gonna have to batch-convert those at some point.

Reply:
I had tried adding it directly to the map a couple of different ways, but it always complained about the alignment. I couldn't find a way to do byte reads, and always got some variation on "wrong read handler" coming from the maincpu component. I found the delegates as a method while looking at other drivers for examples.

Can you suggest a method for putting it directly in the map to work with single byte reads? Or point me to an existing driver that does something similar? I admit I haven't done tons of MAME development, so I'm mainly looking at existing code for examples.

Reply:
If it's an 8-bit port then you'd need a lambda over a `required_ioport`, cfr. this example:
https://github.com/mamedev/mame/blob/9eab73cc4c6b901e91490f075615d7e1b8bc6641/src/mame/capcom/alien.cpp#L186

Reply:
Implemented directly into the memory map as a lambda. Includes a check against the HW_TYPE machine configuration to accurately reflect standard (final production) hardware.